dgram: clean up private APIs

cjihrig opened the pull request:
##### Checklist
- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

apapirovski commented on lib/dgram.js:
```diff
@@ -63,88 +63,37 @@ const errnoException = errors.errnoException;
-  this._receiving = false;
-  this._bindState = BIND_STATE_UNBOUND;
-  this[async_id_symbol] = this._handle.getAsyncId();
+  state.handle = handle;
```
It feels a bit cleaner if these were all declared directly on the `state` object. Meaning:

```js
const state = {
  handle,
  receiving: false,
  ...etc,
};
```

jasnell commented on lib/dgram.js:
```diff
@@ -710,8 +682,74 @@ Socket.prototype.getSendBufferSize = function() {
+// Deprecated private APIs.
+Object.defineProperty(Socket.prototype, '_handle', {
+  get: util.deprecate(function() {
+    console.log(this);
```
why is the `console.log(this)` added?

cjihrig replied:
Because I forgot to remove it *facepalm*

cjihrig commented:
Addressed nits. Backed out the deprecations, and instead added getters/setters so that the PR is no longer semver-major. I'm going to open a follow up semver-major PR that reimplements the deprecations only. For now, this shouldn't cause problems for other PRs.

CI: https://ci.nodejs.org/job/node-test-pull-request/16043/

cjihrig commented:
Landed in d497ebbf9c7b5b66d355e9645b0ab0b2c61527b2, 79e41cc7b071228b22906b8d44d3c9c649a9628b, 3f6ee7578103fc6f25a0711d2df98ebf39c6620d, and 045b07d51833076ea0e4b2f3d7cce94baaa550a5.
